phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java

```diff
@@ -0,0 +1,261 @@
+        }
+    }
+
+    /**
+     * Task runs periodically to clean up task of child views whose parent is dropped
+     *
+     */
+    public static class DropChildViewsTask extends TimerTask {
+        private RegionCoprocessorEnvironment env;
+        private long timeMaxInterval;
+
+        public DropChildViewsTask(RegionCoprocessorEnvironment env, long timeMaxInterval) {
+            this.env = env;
+            this.timeMaxInterval = timeMaxInterval;
+        }
+  
+        @Override
+        public void run() {
+            PhoenixConnection connForTask = null;
+            Timestamp timestamp = null;
+            String tenantId = null;
+            byte[] tenantIdBytes;
+            String schemaName= null;
+            byte[] schemaNameBytes;
+            String tableName = null;
+            byte[] tableNameBytes;
+            PhoenixConnection pconn;
+            try {
+                String taskQuery = "SELECT " +
+                        PhoenixDatabaseMetaData.TASK_TS + ", " +
+                        PhoenixDatabaseMetaData.TENANT_ID + ", " +
+                        PhoenixDatabaseMetaData.TABLE_SCHEM + ", " +
+                        PhoenixDatabaseMetaData.TABLE_NAME +
+                        " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME +
+                        " WHERE "+ PhoenixDatabaseMetaData.TASK_TYPE + " = " + PTable.TaskType.DROP_CHILD_VIEWS.getSerializedValue();
+
+                connForTask = QueryUtil.getConnectionOnServer(env.getConfiguration()).unwrap(PhoenixConnection.class);
+                PreparedStatement taskStatement = connForTask.prepareStatement(taskQuery);
+                ResultSet rs = taskStatement.executeQuery();
+                while (rs.next()) {
+                    try {
+                        // delete child views only if the parent table is deleted from the system catalog
+                        timestamp = rs.getTimestamp(1);
+                        tenantId = rs.getString(2);
+                        tenantIdBytes= rs.getBytes(2);
+                        schemaName= rs.getString(3);
+                        schemaNameBytes = rs.getBytes(3);
+                        tableName= rs.getString(4);
+                        tableNameBytes = rs.getBytes(4);
+
+                        if (tenantId != null) {
+                            Properties tenantProps = new Properties();
+                            tenantProps.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, tenantId);
+                            pconn = QueryUtil.getConnectionOnServer(tenantProps, env.getConfiguration()).unwrap(PhoenixConnection.class);
+
+                        }
+                        else {
+                            pconn = QueryUtil.getConnectionOnServer(env.getConfiguration()).unwrap(PhoenixConnection.class);
+                        }
+
+                        MetaDataProtocol.MetaDataMutationResult result = new MetaDataClient(pconn).updateCache(pconn.getTenantId(),
```

Comment:
Why are we making a new MetaDataClient per result?  Can't we make it per pconn?

Reply:
@dbwong, I do not think we should be concerned about making a newMetaDataClient per result since the objects of this class have only one instance variable which is a pointer to the PhoenixConnection object that is passed via the constructor. Trying to make one MetaDataClient object per connection is probably as expensive as creating a new object per result. I think it will be over engineering if we do that.